src/main/java/com/faforever/client/i18n/I18nImpl.java

```diff
@@ -1,30 +1,61 @@
 package com.faforever.client.i18n;
 
+import com.faforever.client.preferences.LanguageInfo;
+import com.faforever.client.preferences.PreferencesService;
+import org.slf4j.LoggerFactory;
 import org.springframework.context.MessageSource;
+import org.springframework.stereotype.Service;
 
+import javax.annotation.PostConstruct;
 import javax.inject.Inject;
+import java.lang.invoke.MethodHandles;
 import java.util.Locale;
 
+@Service
 public class I18nImpl implements I18n {
 
+  private final org.slf4j.Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private MessageSource messageSource;
+  private Locale locale;
   @Inject
-  MessageSource messageSource;
+  private PreferencesService preferencesService;
+  private Locale userSpecificLocale;
 
   @Inject
-  Locale locale;
+  public I18nImpl(Locale locale, MessageSource messageSource) {
+    this.messageSource = messageSource;
+    this.locale = locale;
+  }
+
+  @PostConstruct
+  public void init() {
+    if (!preferencesService.getPreferences().getLanguage().getLanguage().equals(LanguageInfo.AUTO.getLanguageCode())) {
+      try {
+        userSpecificLocale = new Locale(preferencesService.getPreferences().getLanguage().getLanguage(), LanguageInfo.getLanuageInfoByLanguageCode(preferencesService.getPreferences().getLanguage().getLanguage()).getCountryCode());
```

Comment:
Extract repetitive method calls like `preferencesService.getPreferences().getLanguage().getLanguage()` to a local variable
(Select it, `CTRL+Alt+V`)